### PR TITLE
Add libdeflate

### PR DESCRIPTION
**Pull request by thofma**

### CODE REVIEW

File: L/libdeflate/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "libdeflate"
+version = v"1.6.0"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/ebiggers/libdeflate/archive/v1.6.tar.gz", "60748f3f7b22dae846bc489b22a4f1b75eab052bf403dd8e16c8279f16f5171e"),
+    "./bundled",
```
Comment (giordano) [inline L/libdeflate/build_tarballs.jl:11]:
```suggestion
    DirectorySource("./bundled"),
```

File: L/libdeflate/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd libdeflate-1.6/
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/fix_makefile.patch
+make PROG_SUFFIX=$exeext PREFIX=${prefix} LIBDIR=${libdir} DISABLE_ZLIB=true install
+make PROG_SUFFIX=$exeext PREFIX=${prefix} install
```
Comment (giordano) [inline L/libdeflate/build_tarballs.jl:20]:
```suggestion
make PROG_SUFFIX=$exeext PREFIX=${prefix} LIBDIR=${libdir} install
```

Comment (giordano) [reply L/libdeflate/build_tarballs.jl:20]: Why there are two `make install`?

Comment (thofma) [reply L/libdeflate/build_tarballs.jl]: Oops. Fixed!

File: L/libdeflate/bundled/patches/fix_makefile.patch [hunk 3]
```diff
@@ -0,0 +1,27 @@
+From 2fa25d0f8eee2a467d17dc92f5b25285a8582c74 Mon Sep 17 00:00:00 2001
```
Comment (giordano) [inline L/libdeflate/bundled/patches/fix_makefile.patch:1]: I encourage you submitting the patch upstream :slightly_smiling_face:

Comment (thofma) [reply L/libdeflate/bundled/patches/fix_makefile.patch:1]: Will do. What is the correct way to change the build script (adding the upsteam information about the patch as a comment) without recomputing the binaries?

Comment (giordano) [reply L/libdeflate/bundled/patches/fix_makefile.patch:1]: If you add `[skip ci]` in the commit message, the ci job won't run on master

**Pull request merged**